### PR TITLE
[GraphBolt] init graph topology

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 1]
```diff
@@ -82,14 +88,22 @@ def test_sets(self) -> List[ItemSet] or List[ItemSetDict]:
...
         """Return the feature."""
         return self._feature
 
+    def _load_graph(
+        self, graph_topology: OnDiskGraphTopology
+    ) -> CSCSamplingGraph:
+        """Load the graph topology."""
+        if graph_topology is None:
+            return None
+        return load_csc_sampling_graph(graph_topology.path)
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:105]: In yaml, we have type: CSCSamplingGraph, we should check the type first, such that in the future when we support new type, the old dgl version can failure properly.

File: python/dgl/graphbolt/impl/ondisk_dataset.py [hunk 2]
```diff
@@ -82,14 +88,22 @@ def test_sets(self) -> List[ItemSet] or List[ItemSetDict]:
         """Return the test set."""
         return self._test_sets
 
-    def graph(self) -> object:
+    def graph(self) -> CSCSamplingGraph:
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/ondisk_dataset.py:91]: I think we can still be very soft here to keep it as object.

File: python/dgl/graphbolt/impl/ondisk_dataset.py (hunk 1 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:105]: when we support new type, the old one(type) is supposed to work well?
pydantic is helping us verify the enum value automatically.

File: python/dgl/graphbolt/impl/ondisk_dataset.py (hunk 2 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:91]: sure as more types might be supported later

File: python/dgl/graphbolt/impl/ondisk_dataset.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:105]: I see, the I have less concern. nit: still worth to check the OnDiskGraphTopologyType = CSC_SAMPLING before using load_csc_sampling_graph(graph_topology.path), and return none for others.

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/impl/ondisk_dataset.py (hunk 1 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/ondisk_dataset.py:105]: sure.

**Pull request merged**